llap-server/src/java/org/apache/hadoop/hive/llap/shufflehandler/ShuffleHandler.java

```diff
@@ -797,16 +803,17 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent evt)
 
       Map<String, MapOutputInfo> mapOutputInfoMap =
           new HashMap<String, MapOutputInfo>();
-      Channel ch = evt.getChannel();
-
+      Channel ch = ctx.channel();
       // In case of KeepAlive, ensure that timeout handler does not close connection until entire
       // response is written (i.e, response headers + mapOutput).
-      ChannelPipeline pipeline = ch.getPipeline();
+      ChannelPipeline pipeline = ch.pipeline();
       TimeoutHandler timeoutHandler = (TimeoutHandler)pipeline.get(TIMEOUT_HANDLER);
       timeoutHandler.setEnabledTimeout(false);
 
       String user = userRsrc.get(jobId);
-
+      if (keepAliveParam || connectionKeepAliveEnabled){
```

Comment:
good catch :) this is an epic workaround that I haven't been able to figure out, here are some details:
https://issues.apache.org/jira/browse/TEZ-4157?focusedCommentId=17100835&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17100835

(btw: with netty3, we didn't need this)

are you fine with a comment explaining this?

Reply:
Got it, this is helpful but lets make sure this is expected from nettys' side of things before committing -- this would be helpful for the Tez change as well :)

Reply:
okay, in this case I'll have to include some unit tests here (which might be part of tez code already) + create a simple repro to share with netty community

Reply:
Thanks Laszlo! sounds like a plan!